src/H5CX.c

```diff
@@ -1621,12 +1621,18 @@ H5CX_get_vol_wrap_ctx(void **vol_wrap_ctx)
     H5CX_node_t **head      = NULL;    /* Pointer to head of API context list */
     herr_t        ret_value = SUCCEED; /* Return value */
 
-    FUNC_ENTER_NOAPI_NOERR
+    FUNC_ENTER_NOAPI(FAIL)
 
     /* Sanity check */
     HDassert(vol_wrap_ctx);
     head = H5CX_get_my_context(); /* Get the pointer to the head of the API context, for this thread */
-    HDassert(head && *head);
+
+    /* No error is expected at this point.  But in case an application calls H5VLwrap_register
+     * which doesn't reset the API context and there is no context, returns an relevant error here
+     */
+    if (!head || !(*head))
+        HGOTO_ERROR(H5E_CONTEXT, H5E_CANTGET, FAIL,
+                    "the API context isn't available, e.g. when an application calls H5VLwrap_register")
```

Comment:
I think this should be two error checks instead, one for `!head` with an error message such as `H5E_CONTEXT, H5E_UNINITIALIZED, FAIL, "API context is not available"` and one for `!(*head)` with an error message such as `H5E_CONTEXT, H5E_CANTGET, FAIL, "unable to get current API context"`.

Rather than having "when an application calls H5VLwrap_register" in the error message, I think that's something we should document in the doxygen for the function in H5VLconnector_passthru.h. That header file currently doesn't have any doxygen in it, but we should probably add it at least for this function so users are aware that applications shouldn't call it instead of just running into a failure case.

Reply:
I made changes according to your suggestion.